bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableDataSettingsFactory.java

```diff
@@ -0,0 +1,265 @@
+    }
+
+    batchSettingsBuilder
+        .setIsEnabled(bulkOptions.useBulkApi())
+        .setElementCountThreshold(Long.valueOf(bulkOptions.getBulkMaxRowKeyCount()))
+        .setRequestByteThreshold(bulkOptions.getBulkMaxRequestSize())
+        .setFlowControlSettings(flowControlBuilder.build());
+
+    // TODO: implement bulkMutationThrottling & bulkMutationRpcTargetMs, once available
+    builder.bulkMutationsSettings().setSimpleTimeoutNoRetries(
+        ofMillis(options.getCallOptionsConfig().getShortRpcTimeoutMs()));
+  }
+
+  /**
+   * To build sampleRowKey with default Settings based on retries setting.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildSampleRowKeys(Builder builder, BigtableOptions options) {
+    builder.sampleRowKeysSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * To build mutateRowSettings with default Settings based on retries setting.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildMutateRow(Builder builder, BigtableOptions options) {
+    builder.mutateRowSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * To build readRows with default Settings based on retries setting.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildReadRows(Builder builder, BigtableOptions options) {
+    // TODO: set readPartialRowTimeout for watchdog timer, taken BigtableSession#setupWatchdog()
+    builder.readRowsSettings()
+        .setRetrySettings(defaultRetrySettings(options));
+  }
+
+  /**
+   * This method builds ReadModifyWrite with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildReadModifyWrite(Builder builder, long rpcTimeoutMs) {
+    builder.readModifyWriteRowSettings()
+        .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+  }
+
+  /**
+   * This method builds CheckAndMutateRow with no retry configurations
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static void buildCheckAndMutateRow(Builder builder, long rpcTimeoutMs) {
+    builder.checkAndMutateRowSettings()
+        .setSimpleTimeoutNoRetries(ofMillis(rpcTimeoutMs));
+  }
+
+  /**
+   * To create default RetrySettings, for BigtableDataSettings.
+   *
+   * @param builder a {@link BigtableDataSettings.Builder} object.
+   * @param bulkMutation a {@link BulkOptions} object.
+   */
+  private static RetrySettings defaultRetrySettings(BigtableOptions options) {
```

Comment:
Please rename this to something more appropriate. This method doesn't represent default settings. Maybe something like `buildIdempotentRetrySettings`?